### DISCUSSION TITLE
Type the meta function's data and parentData properties as potentially undefined

### DISCUSSION THREAD

**Discussion by andrelandgraf** [remix-run/remix#5522] [Proposals]
The route's meta function runs even if a loader throws an error. In such cases, the data property is undefined. Currently, the data property is typed as `any`.

We mostly access loader data from the route's component, where it is always defined. In meta functions, it can be undefined, but the current types don't highlight this.

It's too easy to forget about this during development. It would be great if Remix could highlight that the `data` property can be undefined by making it optional or typing it as `any | undefined` - or something similar!

**Comment by sergiodxa** [comment]
Doing `any | undefined` will result in `any`

But MetaFunction accepts a generic now I think, so it could be `Generic | undefined`.
